control_plane/src/storage.rs

```diff
@@ -22,6 +23,28 @@ use zenith_utils::connstring::connection_address;
 
 const HTTP_BASE_URL: &str = "http://127.0.0.1:9898/v1";
 
+pub trait ResponseErrorMessageExt {
+    fn error_from_error_body(self) -> Result<Self>
+    where
+        Self: Sized;
+}
```

Comment:
Since this is an extension method, could you move `Sized` to `ResponoseErrorMessageExt: Sized`? This would be shorter. Also, could you give it a shorter name with less repetition? E.g. `validate_body`.